test-vectors.md

```diff
@@ -38,15 +38,25 @@ the syntax of the messages used for MLS (independent of semantics).
 * `optional<type>` is serialized as the value itself or `null` if not present.
 * MLS structs are binary encoded according to spec and represented as
   hex-encoded strings in JSON.
-* HPKE public keys and signature public keys are encoded in the formats
-  described in the MLS specification.
-* HPKE private keys are encoded according to the `SerializePrivateKey` function
-  for the HPKE method for the ciphersuite.
-* Signature private keys are encoded similarly:
+* HPKE and Signature public keys are encoded as HPKEPublicKey and
+  SignaturePublicKey, as described in the MLS specification.
+* HPKE and Signature private keys are encoded similarly, as TLS-encoded vectors,
+  containing an encoded private key
+  * HPKE private keys are encoded according to the `SerializePrivateKey` function
+    for the HPKE method for the ciphersuite
   * ECDSA private keys are encoded using the `Field-Element-to-Octet-String`
-    transofrmation (i.e., as big-endian integers)
+    transformation (i.e., big-endian integers, as with HPKE ECDH private keys)
   * EdDSA private keys are encoded in their native byte string representation.
 
+
+```
+opaque HPKEPublicKey<V>;
```

Comment:
Shouldn't we have `opaque Signature<V>;` as well? The test vector currently doesn't do this and the description here doesn't say anything about the signature encoding.

Reply:
You make a compelling case for having these all just be raw bytes, without the length prefix :)  I can patch in that direction.

Reply:
I'd be fine with both as long as it's consistent ;) But using raw bytes everywhere might indeed be simpler.

Reply:
Implemented in the latest commit.